src/ui/components/UploadScreen/TeamSelect.tsx

```diff
@@ -25,15 +26,22 @@ export default function TeamSelect({
   selectedWorkspaceId: string;
   handleWorkspaceSelect: (id: string) => void;
 }) {
+  const userInfo = useGetUserInfo();
   const [expanded, setExpanded] = useState(false);
-  const displayedWorkspaces = [personalWorkspace, ...workspaces].sort();
+  const displayedWorkspaces: { id: string; name: string }[] = [...workspaces].sort();
+
+  if (userInfo.features.library) {
+    displayedWorkspaces.unshift(personalWorkspace);
+  }
```

Comment:
this is a nit, but I'd probably do 

```js
displayedWorkspaces = [personalWorkspace, ...displayedWorkspaces]
```

since who can remember whether shift or unshift adds or removes :)